msal/src/main/java/com/microsoft/identity/client/MultipleAccountPublicClientApplication.java

```diff
@@ -239,11 +241,14 @@ public void onError(final Exception exception) {
     public void removeAccount(@Nullable final IAccount account,
                               @NonNull final RemoveAccountCallback callback) {
         ApiDispatcher.initializeDiagnosticContext();
+
+        // First, cast the input IAccount to a MultiTenantAccount
+        final MultiTenantAccount multiTenantAccount = (MultiTenantAccount) account;
+
         //create the parameter
         try {
-            if (null == account
-                    || null == account.getHomeAccountIdentifier()
-                    || StringUtil.isEmpty(account.getHomeAccountIdentifier().getIdentifier())) {
+            if (null == multiTenantAccount
+                    || null == multiTenantAccount.getHomeAccountId()) {
```

Comment:
Should we do `TextUtils.isEmpty()` check as well?

Reply:
Sure -- I don't see a way we could end up with `""` as our value, but for consistency with other parts of the code that evaluate Strings I'll make this change

Reply:
Oh, nevermind. I've made changes to assert that this can't be null. Resolved in e8899e5832120f1468de9c913d2084ad7b8101f1